java/ql/src/semmle/code/java/frameworks/android/Intent.qll

```diff
@@ -27,10 +27,44 @@ class AndroidReceiveIntentMethod extends Method {
   }
 }
 
-class ContextStartActivityMethod extends Method {
-  ContextStartActivityMethod() {
-    (hasName("startActivity") or hasName("startActivities")) and
-    getDeclaringType() instanceof TypeContext
+/** A constructor of `android.content.Intent`. */
+class IntentConstructor extends TaintPreservingCallable {
```

Comment:
We should prefer using CSV style to `TaintPreservingCallable` where practical: it's more verbose, but it exhibits the style we want external contributors to use and it feeds into automatic coverage reports etc.

Reply:
You are right. I was conflicted about whether to maintain the style in `Intent.qll` or to do it in CSV style for new models, but it's better to do it well now that the library is still manageable. I changed this in d400745 following your suggestion, and also I'll be opening another PR changing the rest of the models to CSV. Thanks!